src/circuit/runtime.rs

```diff
@@ -292,12 +292,9 @@ mod tests {
             let root = Root::build_with_scheduler::<_, S>(move |circuit| {
                 let rtclone = runtime.clone();
                 // Generator that produces values using `sequence_next`.
-                let source =
-                    circuit.add_source(Generator::new(move || rtclone.sequence_next(index)));
-                circuit.add_sink(
-                    Inspect::new(move |n: &usize| data_clone.borrow_mut().push(*n)),
-                    &source,
-                );
+                circuit
+                    .add_source(Generator::new(move || rtclone.sequence_next(index)))
```

Comment:
so why not a "generator" API too?
I regard Generator and Inspect as debugging tools, not as nodes that you normally want in circuits.

Reply:
Both `inspect` and generator can do useful work.  We can also add a convenience method for generators.  The difference is that `inspect` is a method of a stream.  `generator` would have to be a method of `Circuit`, because it's a source operator, so there's no stream we can apply it to.